src/FFmpegWriter.cpp

```diff
@@ -910,60 +914,63 @@ void FFmpegWriter::flush_encoders() {
-
-			AVPacket pkt;
-			av_init_packet(&pkt);
-			pkt.data = NULL;
-			pkt.size = 0;
-			pkt.pts = pkt.dts = write_audio_count;
+	if (info.has_audio) {
+        for (;;) {
+            AVPacket pkt;
+            av_init_packet(&pkt);
+            pkt.data = NULL;
+            pkt.size = 0;
+            pkt.pts = pkt.dts = write_audio_count;
 
-			/* encode the image */
-			int error_code = 0;
-			int got_packet = 0;
+            /* encode the image */
+            int error_code = 0;
+            int got_packet = 0;
 #if IS_FFMPEG_3_2
-			error_code = avcodec_send_frame(audio_codec_ctx, NULL);
+            error_code = avcodec_send_frame(audio_codec_ctx, NULL);
 #else
-			error_code = avcodec_encode_audio2(audio_codec_ctx, &pkt, NULL, &got_packet);
+            error_code = avcodec_encode_audio2(audio_codec_ctx, &pkt, NULL, &got_packet);
 #endif
-			if (error_code < 0) {
-				ZmqLogger::Instance()->AppendDebugMethod("FFmpegWriter::flush_encoders ERROR [" + (std::string)av_err2str(error_code) + "]", "error_code", error_code);
-			}
-			if (!got_packet) {
-				break;
-			}
+            if (error_code < 0) {
+                ZmqLogger::Instance()->AppendDebugMethod(
+                        "FFmpegWriter::flush_encoders ERROR [" + (std::string) av_err2str(error_code) + "]",
+                        "error_code", error_code);
+            }
+            if (!got_packet) {
+                break;
+            }
 
-			// Since the PTS can change during encoding, set the value again.  This seems like a huge hack,
-			// but it fixes lots of PTS related issues when I do this.
-			pkt.pts = pkt.dts = write_audio_count;
+            // Since the PTS can change during encoding, set the value again.  This seems like a huge hack,
+            // but it fixes lots of PTS related issues when I do this.
+            pkt.pts = pkt.dts = write_audio_count;
 
-			// Scale the PTS to the audio stream timebase (which is sometimes different than the codec's timebase)
-			if (pkt.pts != AV_NOPTS_VALUE)
-				pkt.pts = av_rescale_q(pkt.pts, audio_codec_ctx->time_base, audio_st->time_base);
-			if (pkt.dts != AV_NOPTS_VALUE)
-				pkt.dts = av_rescale_q(pkt.dts, audio_codec_ctx->time_base, audio_st->time_base);
-			if (pkt.duration > 0)
-				pkt.duration = av_rescale_q(pkt.duration, audio_codec_ctx->time_base, audio_st->time_base);
+            // Scale the PTS to the audio stream timebase (which is sometimes different than the codec's timebase)
+            if (pkt.pts != AV_NOPTS_VALUE)
+                pkt.pts = av_rescale_q(pkt.pts, audio_codec_ctx->time_base, audio_st->time_base);
+            if (pkt.dts != AV_NOPTS_VALUE)
+                pkt.dts = av_rescale_q(pkt.dts, audio_codec_ctx->time_base, audio_st->time_base);
+            if (pkt.duration > 0)
+                pkt.duration = av_rescale_q(pkt.duration, audio_codec_ctx->time_base, audio_st->time_base);
```

Comment:
If we have a `pkt.duration`, seems like that could be used to increment `write_video_count`. As [the documentation says](https://ffmpeg.org/doxygen/4.1/structAVPacket.html#a622e758be29fd500aed0ffdc069550f7), `AVPacket::duration` "Equals next_pts - this_pts in presentation order." (Also, pretty sure `av_packet_rescale_ts()` is a better choice than `av_rescale_q()` here, too.)